### ISSUE TITLE
Resolver error: t.filter is not a function

### ISSUE THREAD

**Issue by EmbeddedBacon** [microsoft/vscode-remote-release#3226]
- VSCode Version: 1.46
- OS Version: Windows 10

Hello,

  When I try connecting to a remote machine and once connected, if I try to open a folder I see the following:

```
[16:59:54.017] Log Level: 2
[16:59:54.018] remote-ssh@0.51.0
[16:59:54.018] win32 x64
[16:59:54.021] SSH Resolver called for "ssh-remote+remote-machine-ip", attempt 1
[16:59:54.022] SSH Resolver called for host: remote-machine-ip
[16:59:54.022] Setting up SSH remote "remote-machine-ip"
[16:59:54.038] Using commit id "a5d1cc28bb5da32ec67e86cc50f84c67cc690321" and quality "stable" for server
[16:59:54.039] Install and start server if needed
[16:59:54.050] Checking ssh with "ssh -V"
[16:59:54.056] Got error from ssh: spawn ssh ENOENT
[16:59:54.056] Checking ssh with "C:\WINDOWS\System32\OpenSSH\ssh.exe -V"
[16:59:54.059] Got error from ssh: spawn C:\WINDOWS\System32\OpenSSH\ssh.exe ENOENT
[16:59:54.059] Checking ssh with "C:\Program Files\Git\usr\bin\ssh.exe -V"
[16:59:54.427] > OpenSSH_7.9p1, OpenSSL 1.1.1a  20 Nov 2018
[16:59:54.441] Running script with connection command: "C:\Program Files\Git\usr\bin\ssh.exe" -T -D 53733 remote-machine-ip bash
[16:59:54.455] Terminal shell path: C:\WINDOWS\System32\cmd.exe
[16:59:55.109] > 
[16:59:55.111] Got some output, clearing connection timeout
[16:59:56.392] > ddf204e12b98: running
> 
[16:59:56.412] > Acquiring lock on /users/user_id/.vscode-server/bin/a5d1cc28bb5da32ec67e86cc50f84c67cc690321/vscode-remote-lock.user_id.a5d1cc28bb5da32ec67e86cc50f84c67cc690321
> \ln /users/user_id/.vscode-server/bin/a5d1cc28bb5da32ec67e86cc50f84c67cc690321/vscode-remote-lock.user_id.a5d1cc28bb5da32ec67e86cc50f84c67cc690321.target /users/user_id/.vscode-server/bin/a5d1cc28bb5da32ec67e86cc50f84c67cc690321/vscode-re
> mote-lock.user_id.a5d1cc28bb5da32ec67e86cc50f84c67cc690321
> Found existing installation at /users/user_id/.vscode-server/bin/a5d1cc28bb5da32ec67e86cc50f84c67cc690321...
> 
[16:59:56.446] > Found running server...
> 
> *
> * Reminder: You may only use this software with Visual Studio family products,
> * as described in the license (https://go.microsoft.com/fwlink/?linkid=2077057)
> *
> 
> 
[16:59:56.466] > Checking server status on port 40343 with wget
> ddf204e12b98: start
> sshAuthSock====
> agentPort==40343==
> osReleaseId==centos==
> arch==x86_64==
> webUiAccessToken====
> tmpDir==/run/user/1000==
> platform==linux==
[16:59:56.513] > 
> ddf204e12b98: end
> 
[16:59:56.513] Received install output: 
sshAuthSock====
agentPort==40343==
osReleaseId==centos==
arch==x86_64==
webUiAccessToken====
tmpDir==/run/user/1000==
platform==linux==
 
[16:59:56.515] Remote server is listening on port 40343
[16:59:56.516] Parsed server configuration: {"agentPort":40343,"osReleaseId":"centos","arch":"x86_64","webUiAccessToken":"","sshAuthSock":"","tmpDir":"/run/user/1000","platform":"linux"}
[16:59:56.523] Starting forwarding server. localPort 53735 -> socksPort 53733 -> remotePort 40343
[16:59:56.526] Forwarding server listening on 53735
[16:59:56.526] Waiting for ssh tunnel to be ready
[16:59:56.530] Tunneled remote port 40343 to local port 53735
[16:59:56.531] Resolved "ssh-remote+remote-machine-ip" to "127.0.0.1:53735"
[16:59:56.534] [Forwarding server 53735] Got connection 0
[16:59:56.558] Resolver error: t.filter is not a function
[16:59:56.594] ------
```

I have tried deleting .vscode-server, but that didn't see to help.  I was searching around and found instances of "t.filter" in some javascript files.   I am able to connect to another remote machine and open a directory without an issues.  

Mark

**Comment by roblourens** [comment]
I am looking at the code and can't figure out where this is coming from. There are a bunch of `t.filter` but none seem right. I'm pushing a change to log the full stack. I'd appreciate if you install the "nightly" version of the ssh extension instead and try it out there, and post the log again. Thanks!

**Comment by roblourens** [comment]
Actually you don't have to try the nightly build yet, I am having issues publishing it right now. Will update...

**Comment by EmbeddedBacon** [comment]
@roblourens, thanks, sir

**Comment by roblourens** [comment]
Ok, a new nightly version is published, would appreciate seeing the log with this new version.

**Comment by EmbeddedBacon** [comment]
Mr. @roblourens, how do I get the nightly version?  I have `C_Cpp.updateChannel` set to `Insiders`.  I didn't see anything else for "Install Another Version".  How do I install the nightly version?  Thanks.

**Comment by hugo-chan** [comment]
Had the same issue. Using Nightly version works. @EmbeddedBacon, it's another extension

**Comment by EmbeddedBacon** [comment]
@hugo-chan   Thanks, found it.  Will give it a try

**Comment by EmbeddedBacon** [comment]
@roblourens, disabled the "Remote - SSH" extension, installed "Remote - SSH (Nightly)" and don't see the error now.  Appears to be fixed, at least in the nightly.

**Comment by EmbeddedBacon** [comment]
Here is the captured log, scrubbed for personal info

`[19:27:22.341] Log Level: 2
[19:27:22.342] remote-ssh-nightly@2020.6.31049
[19:27:22.342] win32 x64
[19:27:22.346] SSH Resolver called for "ssh-remote+machine_hostname", attempt 1
[19:27:22.348] SSH Resolver called for host: machine_hostname
[19:27:22.348] Setting up SSH remote "machine_hostname"
[19:27:22.369] Using commit id "cd9ea6488829f560dc949a8b2fb789f3cdc05f5d" and quality "stable" for server
[19:27:22.371] Install and start server if needed
[19:27:22.385] Checking ssh with "ssh -V"
[19:27:22.393] Got error from ssh: spawn ssh ENOENT
[19:27:22.396] Checking ssh with "C:\WINDOWS\System32\OpenSSH\ssh.exe -V"
[19:27:22.399] Got error from ssh: spawn C:\WINDOWS\System32\OpenSSH\ssh.exe ENOENT
[19:27:22.400] Checking ssh with "C:\Program Files\Git\usr\bin\ssh.exe -V"
[19:27:22.909] > OpenSSH_7.9p1, OpenSSL 1.1.1a  20 Nov 2018
[19:27:22.919] Running script with connection command: "C:\Program Files\Git\usr\bin\ssh.exe" -T -D 61517 machine_hostname bash
[19:27:22.925] Terminal shell path: C:\WINDOWS\System32\cmd.exe
[19:27:23.552] > 
[19:27:23.553] Got some output, clearing connection timeout
[19:27:25.594] > user_name@machine_name's password:
[19:27:31.214] > 
> 
[19:27:31.517] > af507b1b533d: running
[19:27:31.576] > 
> Acquiring lock on /home/user_name/.vscode-server/bin/cd9ea6488829f560dc949a8b2fb789f3cdc05f5d/vscode-remote-lock.user_name.cd9ea6488829f560dc949a8b2fb789f3cdc05f5d
> \ln /home/user_name/.vscode-server/bin/cd9ea6488829f560dc949a8b2fb789f3cdc05f5d/vscode-remote-lock.user_name.cd9ea6488829f560dc949a8b2fb789f3cdc05f5d.target /home/user_
> name/.vscode-server/bin/cd9ea6488829f560dc949a8b2fb789f3cdc05f5d/vscode-remote-lock.user_name.cd9ea6488829f560dc949a8b2fb789f3cdc05f5d
> 
[19:27:31.583] > Found existing installation at /home/user_name/.vscode-server/bin/cd9ea6488829f560dc949a8b2fb789f3cdc05f5d...
> Found running server...
> 
> *
> * Reminder: You may only use this software with Visual Studio family products,
> * as described in the license (https://go.microsoft.com/fwlink/?linkid=2077057)
> *
> 
> 
[19:27:31.627] > Checking server status on port 40209 with wget
> af507b1b533d: start
> sshAuthSock====
> agentPort==40209==
> osReleaseId==centos==
> arch==x86_64==
> webUiAccessToken====
> tmpDir==/run/user/1205925==
> platform==linux==
> af507b1b533d: end
> 
[19:27:31.628] Received install output: 
sshAuthSock====
agentPort==40209==
osReleaseId==centos==
arch==x86_64==
webUiAccessToken====
tmpDir==/run/user/1205925==
platform==linux==
 
[19:27:31.629] Remote server is listening on port 40209
[19:27:31.630] Parsed server configuration: {"agentPort":40209,"osReleaseId":"centos","arch":"x86_64","webUiAccessToken":"","sshAuthSock":"","tmpDir":"/run/user/1205925","platform":"linux"}
[19:27:31.636] Starting forwarding server. localPort 61523 -> socksPort 61517 -> remotePort 40209
[19:27:31.642] Forwarding server listening on 61523
[19:27:31.643] Waiting for ssh tunnel to be ready
[19:27:31.652] Tunneled remote port 40209 to local port 61523
[19:27:31.652] Resolved "ssh-remote+machine_hostname" to "127.0.0.1:61523"
[19:27:31.655] [Forwarding server 61523] Got connection 0
[19:27:31.680] ------



 
[19:27:31.745] [Forwarding server 61523] Got connection 1
[19:27:31.748] [Forwarding server 61523] Got connection 2
[19:27:34.115] [Forwarding server 61523] Got connection 3
[19:27:45.251] [Forwarding server 61523] Got connection 4
[19:27:45.255] [Forwarding server 61523] Got connection 5
[19:27:45.258] [Forwarding server 61523] Got connection 6
[19:27:45.264] [Forwarding server 61523] Got connection 7
[19:30:31.615] > 
[19:33:31.609] > 
[19:36:31.607] > 
[19:36:42.642] Picking SSH host
[19:36:45.106] Selected machine_hostname`

**Comment by roblourens** [comment]
Yes, that nightly extension is what I meant. Great, I'm not sure why it was fixed but please let me know if it comes up again. Will release that nightly to stable soon.
